packages/grid/x-data-grid/src/internals/hooks/features/rows/useGridRowsMeta.ts

```diff
@@ -44,16 +48,11 @@ export const useGridRowsMeta = (
   }));
 
   const hydrateRowsMeta = React.useCallback(() => {
-    const { rows } = getCurrentPageRows(apiRef, {
-      pagination,
-      paginationMode,
-    });
-
     apiRef.current.setState((state) => {
       const positions: number[] = [];
       const densityFactor = gridDensityFactorSelector(state, apiRef.current.instanceId);
       const currentRowHeight = gridDensityRowHeightSelector(state, apiRef.current.instanceId);
-      const currentPageTotalHeight = rows.reduce((acc: number, row) => {
+      const currentPageTotalHeight = currentPage.rows.reduce((acc: number, row) => {
         positions.push(acc);
```

Comment:
Should the position start with `initialHeights.spacingTop`?

Reply:
I didn't understand. This `positions.push(acc);` is adding the position of the previous row. Note that `positions[0]` is always 0.

Reply:
My idea was the following:
If the first row has a `margin-top: 5px` which in your system correspond to `initialHeights.spacingTop=5` then the firs row position should reflect this margin with `positions[0] = '5px'`

Reply:
With `initialHeights.spacingTop > 0` the `positions` array is more like how many pixels should I scroll to reach the beginning of the top margin of a row. Although visually the margin is outside the row, here it's part of the row so I think makes sense to keep 0 as the initial value. I don't want to change because I also don't know the consequences of this yet. But one problem of not discounting the margin is that when navigating with the keyboard it scroll up to the top margin. Maybe we could have a option in `scrollToIndexes` to control de behavior. #4013 asks for more parameters.